### DISCUSSION TITLE
Remember current position per view too?

### DISCUSSION THREAD

**Discussion by newbthenewbd** [orbitalquark/textadept#323] [Ideas]
Split views constitute a really strong feature of Textadept, and their implementation is very intuitive. Personally, I tend to use them to see multiple areas of the same file simultaneously. Another very nice feature is that the session is saved, so that I can close the editor, and open it back in the same place.

Except, these two features are sort of incompatible with each other, as the position in the file is only saved globally for the whole buffer, and not per view. This means that my editor, with three views of the same file, will open all three in the same place, requiring manual readjustment of two of them every restart.

Now, who restarts their editor that often? I could just wait till it disturbs me enough to alter my perceiving of spacetime, then hastily put something together in my init.lua that improves this for my specific usecase ~~(while breaking all the other)~~. But I wonder - if enough folks value the "even of the same file" part of unlimited split views, maybe a proper solution could be aimed at? :)

The initial idea that I haven't deeply thought about yet, but that seems like it could work, would be to save to the session file as many current positions for each buffer as there are views, structured just like the views, getting the existing positions duplicated in the background when new views are made, just like the views onscreen have it right now in-memory... Or did somebody perhaps already write a module to do that? :smile:

**Comment by orbitalquark** [comment]
I am not aware of a module that does it. This has been on my TODO list for a while, but is a low priority. It is a complicated feature to implement, as view-specific data would be tied to table/memory addresses so saving/loading from session files is non-trivial because cycling between views is non-linear depending on split structure. Feel free to experiment though and let me know what you come up with.
